Sources/Reporters/DirectoryTree/DirectoryTreeReporter.swift

```diff
@@ -16,24 +51,55 @@ public struct DirectoryTreesReporter: DiagnosticsReporting {
     let includeHiddenFiles: Bool
     let includeSymbolicLinks: Bool
     let printFullPath: Bool
+    let trunks: [Directory]
 
     /// Creates a new reporter for directory trees.
+    /// eventually uses the new init(trunk: [Directory])
     /// - Parameters:
     ///   - directories: The directories to create trees for.
     ///   - maxDepth: The max depth of directories to visit. Defaults to `.max`.
     ///   - maxLength: The max length of nodes in a directory to handle. Defaults to `10`.
     ///   - includeHiddenFiles: Whether hidden files should be captured. Defaults to `false`.
     ///   - includeSymbolicLinks: Whether symbolic links should be captured. Defaults to `false`.
     ///   - printFullPath: Whether the full path of the node should be printed or just the name. Defaults to `false`.
+    @available(*, deprecated, message: "Use the new init(trunks: [Directory])")
     public init(
         directories: [URL],
         maxDepth: Int = .max,
         maxLength: Int = 10,
         includeHiddenFiles: Bool = false,
         includeSymbolicLinks: Bool = false,
         printFullPath: Bool = false
+    ) {
+        var newTrunks: [Directory] = []
+        directories.forEach { directoryURL in
+            newTrunks.append(Directory(url: directoryURL,
+                                    maxDepth: maxDepth,
+                                    maxLength: maxLength,
+                                    includeHiddenFiles: includeHiddenFiles,
+                                    includeSymbolicLinks: includeSymbolicLinks,
+                                    printFullPath: printFullPath))
+        }
+        self.init(directories: directories, trunks: newTrunks)
+    }
+    
+    /// Creates a new reporter for directory trees
+    /// - Parameter trunks: Array Collection of <Directory> to be diagnosed
+    public init(trunks: [Directory]) {
```

Comment:
This initalizer will be irrelevant after removing the `directories` property

Reply:
yep yep, 
gotcha 🙂